src/components/style/interactive/button.css

```diff
@@ -0,0 +1,192 @@
+  --button-secondary-pressed-color: #525355;
+  --button-secondary-destructive: #DA4544;
+  --button-secondary-destructive-hover: #F0EDEF;
+  --button-secondary-destructive-pressed: #ECD6D9;
+  border: 0 none;
+  letter-spacing: 1px;
+  position: relative;
+  line-height: normal;
+  user-select: none;
+  /* We do this to make sure buttons with borders and without are the same dimensions */
+  border: 1px solid transparent;
+  /* we remove 1px off each side as there's a border we need to compensate for */
+  padding: calc(var(--space-half) - 1px) calc(var(--space-default) - 1px);
+  transition: background-color var(--button-group-timeout) var(--motion-easing-default),
+    box-shadow var(--button-group-timeout) var(--motion-easing-default),;
+}
+
+.fdsButton--block {
+  width: 100%;
+}
+
+.fdsButton--loading,
+.fdsButton--disabled {
+  opacity: 0.6;
+  pointer-events: none;
+}
+
+/*
+   Spinner technique is from the following:
+   https://stephanwagner.me/only-css-loading-spinner
+   https://codepen.io/stuartjnelson/pen/zjveOO
+*/
+@keyframes spinner {
+  0% {
+    transform: translate3d(-50%, -50%, 0) rotate(0deg);
+  }
+
+  100% {
+    transform: translate3d(-50%, -50%, 0) rotate(360deg);
+  }
+}
+
+.fdsButton--loading::before {
+  content: "";
+  box-sizing: border-box;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  width: 16px;
+  height: 16px;
+  border-radius: 50%;
+  border: 2px solid var(--color-white);
+  border-top-color: transparent;
+  animation: spinner 600ms linear infinite;
+  transform: translate3d(-50%, -50%, 0);
+  will-change: transform;
+}
+
+.fdsButton--hidden {
+  visibility: hidden;
+}
+
+.fdsButton:focus {
+  outline: 0;
+  box-shadow: 0 0 0 2px #B7D7E5;
+}
+
+.fdsButton--blue {
+  color: var(--color-white);
+  background-color: var(--blue);
+}
+
+.fdsButton--blue:hover {
+  background-color: var(--blue-hover);
+}
+
+.fdsButton--blue:active {
+  background-color: var(--blue-pressed);
+}
+
+.fdsButton--blue.fdsButton--isDestructive {
+  background-color: var(--destructive);
+}
+
+.fdsButton--blue.fdsButton--isDestructive:hover {
+  background-color: var(--destructive-hover);
+}
+
+.fdsButton--blue.fdsButton--isDestructive:active {
+  background-color: var(--destructive-pressed);
+}
+
+.fdsButton--outlined {
+  border: 1px solid var(--button-outlined-border);
+}
+
+.fdsButton--ghost,
+.fdsButton--outlined {
+  /* this isn't needed, just writing out for clarity */
+  color: var(--button-secondary-color);
+}
+
+.fdsButton--outlined:hover,
+.fdsButton--ghost:hover {
+  background-color: var(--button-secondary-hover);
+  color: var(--button-secondary-hover-color);
+}
+
+.fdsButton--outlined:active,
+.fdsButton--ghost:active {
+  background-color: var(--button-secondary-pressed);
+  color: var(--button-secondary-pressed-color);
+}
+
+.fdsButton--outlined.fdsButton--isDestructive:hover,
+.fdsButton--ghost.fdsButton--isDestructive:hover {
+  background-color: var(--button-secondary-destructive-hover);
+}
```

Comment:
So you might be thinking here: why don't you do this: 

```
.fdsButton--isDesctructive:matches(.fdsButton--outlined, .fdsButton--ghost) {}
```

And the answer is... it doesn't work straightforward? Something is happening here where what should be the correct order and specificity isn't being preserved.

Reply:
^ incorrect @rey-wright . I've updated with a working example. You're welcome.